plex/api.py

```diff
@@ -663,13 +676,28 @@ def fillVideoInfos(
 
         if mediaPart:
             # extract the absolute / actual path and the stream URL from the selected MediaPart
-            info['path'] = mediaPart.file
-            item.setPath(plexServer.url(mediaPart.key, includeToken=True))
+            info['path'] = mediaPart.file  # path to file on disk
+
+            # determine if directPlay is enabled and possible
+            if allowDirectPlay:
+                log("Configuring item for direct play", xbmc.LOGDEBUG)
+                # perform replacement in file path if configured
+                filePath = mediaPart.file
+
+                # confirm file is accessible, fall back to streaming otherwise
+                if filePath and xbmcvfs.exists(filePath):
+                    item.setPath(filePath)
+                else:
+                    log(f"Direct play configured but path {filePath} is inaccessible.", xbmc.LOGWARNING)
+                    item.setPath(plexServer.url(mediaPart.key, includeToken=True))
+            else:
+                item.setPath(plexServer.url(mediaPart.key, includeToken=True))
         elif isFolder:
             # for folders use locations for the path
             if locations:
                 info['path'] = locations[0]
             item.setPath(plexServer.url(plexItem.key, includeToken=True))
+
```

Comment:
This change looks unrelated. Can you move it into a separate "cosmetics" commit?